### ISSUE TITLE
[minikube docker] AWX namespace and pods missing after restart

### ISSUE THREAD

**Issue by Commifreak** [ansible/tower-operator#768]
I dont know if this is actual an operator issue, so please lead me to the right direction then :)

I set all up as usual and applied the awx-demo.yml. All AWX pods are ready and up.

A simple `minikube stop` and `minikube start` (with provided params) restarts the cluster, but the whole awx namespace is gone! even the pods. But I still can see the postgre volume in /var/lib/docker.

The first time I noticed that behaviour was after an upgrade of minikube (and the cluster itself) a week ago. I restarted the minikube cluster and every awx pod was gone.

```
# minikube version
minikube version: v1.25.1
commit: 3e64b11ed75e56e4898ea85f96b2e4af0301f43d
```

```
# minikube start --cpus=4 --memory=8g --driver=docker --addons=ingress

Starting...
* minikube v1.25.1 on Ubuntu 20.04
* Using the docker driver based on existing profile
! Your cgroup does not allow setting memory.
  - More information: https://docs.docker.com/engine/install/linux-postinstall/#your-kernel-does-not-support-cgroup-swap-limit-capabilities
! Your cgroup does not allow setting memory.
  - More information: https://docs.docker.com/engine/install/linux-postinstall/#your-kernel-does-not-support-cgroup-swap-limit-capabilities
* Starting control plane node minikube in cluster minikube
* Pulling base image ...
* Restarting existing docker container for "minikube" ...
* Found network options:
  - NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.99.0/24,192.168.39.0/24,192.168.49.0/24
  - http_proxy=http://172.31.42.58:3128
  - https_proxy=http://172.31.42.58:3128
! This container is having trouble accessing https://k8s.gcr.io
* To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
* Preparing Kubernetes v1.23.1 on Docker 20.10.12 ...
  - env NO_PROXY=localhost,127.0.0.1,10.96.0.0/12,192.168.99.0/24,192.168.39.0/24,192.168.49.0/24
  - env HTTP_PROXY=http://172.31.42.58:3128
  - env HTTPS_PROXY=http://172.31.42.58:3128
  - kubelet.housekeeping-interval=5m
  - Generating certificates and keys ...
  - Booting up control plane ...
  - Configuring RBAC rules ...
* Verifying Kubernetes components...
  - Using image gcr.io/k8s-minikube/storage-provisioner:v5
  - Using image k8s.gcr.io/ingress-nginx/controller:v1.1.0
  - Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.1.1
  - Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.1.1
* Verifying ingress addon...
* Enabled addons: storage-provisioner, default-storageclass, ingress
* kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

### After 1st time setup of a fresh awx instance

```
minikube kubectl -- get all -A
NAMESPACE       NAME                                                  READY   STATUS      RESTARTS        AGE
awx             pod/awx-9cb94f674-72656                               4/4     Running     0               3m13s
awx             pod/awx-operator-controller-manager-dfc6f67d7-xfkg4   2/2     Running     0               6m38s
awx             pod/awx-postgres-0                                    1/1     Running     0               4m4s
ingress-nginx   pod/ingress-nginx-admission-create-hlvfq              0/1     Completed   0               7m28s
ingress-nginx   pod/ingress-nginx-admission-patch-n6bbm               0/1     Completed   1               7m28s
ingress-nginx   pod/ingress-nginx-controller-6d5f55986b-h5js4         1/1     Running     0               7m28s
kube-system     pod/coredns-64897985d-rvpz6                           1/1     Running     0               7m28s
kube-system     pod/etcd-minikube                                     1/1     Running     0               7m41s
kube-system     pod/kube-apiserver-minikube                           1/1     Running     0               7m43s
kube-system     pod/kube-controller-manager-minikube                  1/1     Running     0               7m43s
kube-system     pod/kube-proxy-qtdcw                                  1/1     Running     0               7m29s
kube-system     pod/kube-scheduler-minikube                           1/1     Running     0               7m41s
kube-system     pod/storage-provisioner                               1/1     Running     1 (6m58s ago)   7m39s

NAMESPACE       NAME                                                      TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                      AGE
awx             service/awx-operator-controller-manager-metrics-service   ClusterIP   10.110.142.100   <none>        8443/TCP                     6m38s
awx             service/awx-postgres                                      ClusterIP   None             <none>        5432/TCP                     4m4s
awx             service/awx-service                                       NodePort    10.107.244.195   <none>        80:30095/TCP                 3m15s
default         service/kubernetes                                        ClusterIP   10.96.0.1        <none>        443/TCP                      7m43s
ingress-nginx   service/ingress-nginx-controller                          NodePort    10.103.214.236   <none>        80:30407/TCP,443:30421/TCP   7m39s
ingress-nginx   service/ingress-nginx-controller-admission                ClusterIP   10.96.73.27      <none>        443/TCP                      7m39s
kube-system     service/kube-dns                                          ClusterIP   10.96.0.10       <none>        53/UDP,53/TCP,9153/TCP       7m41s

NAMESPACE     NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
kube-system   daemonset.apps/kube-proxy   1         1         1       1            1           kubernetes.io/os=linux   7m41s

NAMESPACE       NAME                                              READY   UP-TO-DATE   AVAILABLE   AGE
awx             deployment.apps/awx                               1/1     1            1           3m13s
awx             deployment.apps/awx-operator-controller-manager   1/1     1            1           6m38s
ingress-nginx   deployment.apps/ingress-nginx-controller          1/1     1            1           7m39s
kube-system     deployment.apps/coredns                           1/1     1            1           7m41s

NAMESPACE       NAME                                                        DESIRED   CURRENT   READY   AGE
awx             replicaset.apps/awx-9cb94f674                               1         1         1       3m13s
awx             replicaset.apps/awx-operator-controller-manager-dfc6f67d7   1         1         1       6m38s
ingress-nginx   replicaset.apps/ingress-nginx-controller-6d5f55986b         1         1         1       7m29s
kube-system     replicaset.apps/coredns-64897985d                           1         1         1       7m29s

NAMESPACE   NAME                            READY   AGE
awx         statefulset.apps/awx-postgres   1/1     4m4s

NAMESPACE       NAME                                       COMPLETIONS   DURATION   AGE
ingress-nginx   job.batch/ingress-nginx-admission-create   1/1           7s         7m39s
ingress-nginx   job.batch/ingress-nginx-admission-patch    1/1           8s         7m39s
```

### `minikube stop` and `minikube start --cpus=4 --memory=8g --driver=docker --addons=ingress`

```
minikube kubectl -- get all -A
NAMESPACE       NAME                                            READY   STATUS      RESTARTS       AGE
ingress-nginx   pod/ingress-nginx-admission-create-h4jn2        0/1     Completed   1              3m37s
ingress-nginx   pod/ingress-nginx-admission-patch-qqmtg         0/1     Completed   2              3m37s
ingress-nginx   pod/ingress-nginx-controller-6d5f55986b-w64b8   1/1     Running     0              3m36s
kube-system     pod/coredns-64897985d-ghxwp                     1/1     Running     0              3m36s
kube-system     pod/etcd-minikube                               1/1     Running     1              3m49s
kube-system     pod/kube-apiserver-minikube                     1/1     Running     1              3m49s
kube-system     pod/kube-controller-manager-minikube            1/1     Running     1              3m49s
kube-system     pod/kube-proxy-z4fzl                            1/1     Running     0              3m37s
kube-system     pod/kube-scheduler-minikube                     1/1     Running     1              3m49s
kube-system     pod/storage-provisioner                         1/1     Running     1 (3m6s ago)   3m48s

NAMESPACE       NAME                                         TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
default         service/kubernetes                           ClusterIP   10.96.0.1       <none>        443/TCP                      3m51s
ingress-nginx   service/ingress-nginx-controller             NodePort    10.99.31.21     <none>        80:30913/TCP,443:30064/TCP   3m48s
ingress-nginx   service/ingress-nginx-controller-admission   ClusterIP   10.107.74.179   <none>        443/TCP                      3m48s
kube-system     service/kube-dns                             ClusterIP   10.96.0.10      <none>        53/UDP,53/TCP,9153/TCP       3m50s

NAMESPACE     NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
kube-system   daemonset.apps/kube-proxy   1         1         1       1            1           kubernetes.io/os=linux   3m50s

NAMESPACE       NAME                                       READY   UP-TO-DATE   AVAILABLE   AGE
ingress-nginx   deployment.apps/ingress-nginx-controller   1/1     1            1           3m48s
kube-system     deployment.apps/coredns                    1/1     1            1           3m50s

NAMESPACE       NAME                                                  DESIRED   CURRENT   READY   AGE
ingress-nginx   replicaset.apps/ingress-nginx-controller-6d5f55986b   1         1         1       3m37s
kube-system     replicaset.apps/coredns-64897985d                     1         1         1       3m37s

NAMESPACE       NAME                                       COMPLETIONS   DURATION   AGE
ingress-nginx   job.batch/ingress-nginx-admission-create   1/1           33s        3m48s
ingress-nginx   job.batch/ingress-nginx-admission-patch    1/1           48s        3m48s
```

What could cause this? This was never an issue before.

**Comment by MrCreeper2033** [comment]
I've got the same problem. It seems to be an issue with the new minikube version 1.25.1
There is aleready an issue open --> https://github.com/kubernetes/minikube/issues/13543  

Workaround: downgraded minikube to version 1.24.0

**Comment by nixocio** [comment]
Hello,

This is not an issue with AWX.

If you need help troubleshooting or are looking for help using AWX, try our mailing list or IRC channel:

#ansible-awx on[ https://libera.chat/](https://libera.chat/)
https://groups.google.com/forum/#!forum/awx-project

**Comment by Commifreak** [comment]
So - For the rest of the world:

I've updated minikube with the patched one. My deplyoment is still lost.

But I still have the raw pgdata dir AND an old awx-backup-claim backup (for the secrets - there is sure another way to get the follwong done without them):

I created a completely new awx deployment. The db is empty at this point.

I manually patched the secrets `awx-postgres-configuration` and `awx-secret-key` via `minikube kubectl -- edit xxx --namespace=$NAMESPACE`.

Then I stopped minikube and replaced the raw pgdata files. Started minikube and voila: the db connections was established, some migrations were made and all data is back!
